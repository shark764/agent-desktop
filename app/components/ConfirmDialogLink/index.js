/**
*
* ConfirmDialogLink
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import Radium from 'radium';

import A from 'components/A';
import ConfirmDialog from 'components/ConfirmDialog';

export class ConfirmDialogLink extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showConfirmDialog: false,
    };
  }

  styles = {
    base: {
      position: 'relative',
    },
    confirmDialog: {
      position: 'absolute',
      bottom: '20px',
      left: '-10px',
    },
  };

  render() {
    return (
      <div style={this.styles.base}>
        <A id={this.props.id} disabled={this.props.disabled} onClick={() => this.setState({ showConfirmDialog: !this.state.showConfirmDialog })} text={this.props.linkText} />
        <ConfirmDialog
          leftAction={this.props.leftAction}
          rightAction={this.props.rightAction}
          leftMessage={this.props.leftMessage}
          rightMessage={this.props.rightMessage}
          leftDisabled={this.props.leftDisabled}
          rightDisabled={this.props.rightDisabled}
          isVisible={this.state.showConfirmDialog}
          hide={() => this.setState({ showConfirmDialog: false })}
          style={this.styles.confirmDialog}
        />
      </div>
    );
  }
}

ConfirmDialogLink.propTypes = {
  id: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  linkText: PropTypes.string.isRequired,
  leftMessage: PropTypes.object,
  rightMessage: PropTypes.object,
  leftAction: PropTypes.func.isRequired,
  rightAction: PropTypes.func.isRequired,
  leftDisabled: PropTypes.bool,
  rightDisabled: PropTypes.bool,
};

export default Radium(ConfirmDialogLink);
