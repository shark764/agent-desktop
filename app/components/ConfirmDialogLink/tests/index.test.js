import React from 'react';
import { shallow } from 'enzyme';

import ConfirmDialogLink from '../index';

describe('<ConfirmDialogLink />', () => {
  it('should render correctly', () => {
    const rendered = shallow(
      <ConfirmDialogLink
        id="mockId"
        disabled={false}
        linkText="test link text"
        leftAction={() => {}}
        rightAction={() => {}}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});
