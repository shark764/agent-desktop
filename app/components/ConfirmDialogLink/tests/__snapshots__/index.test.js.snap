// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmDialogLink /> should render correctly 1`] = `
<div
  data-radium={true}
  style={
    Object {
      "position": "relative",
    }
  }
>
  <A
    disabled={false}
    id="mockId"
    onClick={[Function]}
    text="test link text"
  />
  <ConfirmDialog
    hide={[Function]}
    isVisible={false}
    leftAction={[Function]}
    leftMessage={
      Object {
        "defaultMessage": "No",
        "id": "app.components.ConfirmDialog.no",
      }
    }
    rightAction={[Function]}
    rightMessage={
      Object {
        "defaultMessage": "Yes",
        "id": "app.components.ConfirmDialog.yes",
      }
    }
    style={
      Object {
        "bottom": "20px",
        "left": "-10px",
        "position": "absolute",
      }
    }
  />
</div>
`;
